reactor-netty-http/src/main/java/reactor/netty/http/server/DefaultHttpServerRoutes.java

```diff
@@ -104,7 +107,30 @@ public HttpServerRoutes route(Predicate<? super HttpServerRequest> condition,
 			return Mono.error(t); //500
 		}
 
-		return response.sendNotFound();
+		if (matchHandlers.isEmpty()) {
+			return response.sendNotFound();
+		}
+
+		// only one handler match request,just apply for speed
+		if (matchHandlers.size() == 1) {
+			return matchHandlers.get(0).apply(request, response);
+		}
+
+		for (HttpRouteHandler handler : matchHandlers) {
```

Comment:
if the `HttpRouteHandler` itself becomes `Comparable`, maybe we can sort the `this.handlers` collection once set up is done, rather than on each application of the routes?

Reply:
it will be better if `HttpRouteHandler` itself becomes `Comparable` like `HandlerMapping` in spring-web-reactive project.Under this condition `this handlers` just need sort when add new handler.Now,we can also do this.